### PR TITLE
feat: add global search command 

### PR DESCRIPTION
**Pull request by DJKnaeckebrot**
I used the cmdk command overlay to create a search function for projects and services. It is globally available via ctrl/cmd + j 

image

The component queries the projects and services via the api and redirects to the specific page. It can be used with keyboard or mouse :)

### CODE REVIEW

**Comment by 190km** [comment]
Nice that looks pretty, what about a little circle with the status of the service ?

**Comment by Siumauricio** [comment]
Hey Looks great!, however when we have services with the same name it hover all of them

Screenshot 2024-12-10 at 9 59 24 PM

also the @190km suggestions sounds good, can you add some status to the right side?

File: apps/dokploy/components/dashboard/search-command.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+              onSelect={() => {
+                router.push("/dashboard/requests");
+                setOpen(false);
+              }}
+            >
+              Requests
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/profile");
```
Comment (Siumauricio) [inline apps/dokploy/components/dashboard/search-command.tsx]:
```suggestion
                router.push("/dashboard/settings/server");
```

Comment (Siumauricio) [inline apps/dokploy/components/dashboard/search-command.tsx]:
```suggestion
                router.push("/dashboard/settings/profile");
```

File: apps/dokploy/components/dashboard/search-command.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+              onSelect={() => {
+                router.push("/dashboard/traefik");
+                setOpen(false);
+              }}
+            >
+              Traefik
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/docker");
```
Comment (Siumauricio) [inline apps/dokploy/components/dashboard/search-command.tsx]:
```suggestion
                router.push("/dashboard/docker");
```

File: apps/dokploy/components/dashboard/search-command.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            </CommandItem>
+            <CommandItem
+              onSelect={() => {
+                router.push("/dashboard/profile");
+                setOpen(false);
+              }}
+            >
+              Settings
+            </CommandItem>
+          </CommandGroup>
```
Comment (Siumauricio) [inline apps/dokploy/components/dashboard/search-command.tsx]: Can you hide the following items when is the cloud version?

you can use this endpoint

```ts
const { data: isCloud, isLoading } = api.settings.isCloud.useQuery();
```

- Monitoring
- Traefik
- Docker
- Requests

**Review by Siumauricio — CHANGES_REQUESTED** [review]

**Comment by DJKnaeckebrot** [comment]
> Nice that looks pretty, what about a little circle with the status of the service ?

Sure I can see what I can do :)



> Hey Looks great!, however when we have services with the same name it hover all of them
> 
> Screenshot 2024-12-10 at 9 59 24 PM
> 
> also the @190km suggestions sounds good, can you add some status to the right side?

How about making the name project / servicename ? Than it should be clear I guess ? I'll also make it hide the other stuff on cloud setuo :)

**Comment by DJKnaeckebrot** [comment]
image

@190km something like this? 

@Siumauricio Hope that fixed the issue with the same service names as it now should be clear which project it is in :)

**Comment by 190km** [comment]
Looks amazing !

**Comment by Siumauricio** [comment]
Looks awesome now! @DJKnaeckebrot, let me know when you ready to check the changes, Thanks!

**Comment by DJKnaeckebrot** [comment]
I pushed the latest changes, so should be ready for review @Siumauricio 

I added the application.id as hidden text which seems to resolve the issue for now. If anyone ever finds a better solution/workaround would be awesome!

I also implemented the suggestion from @drudge to use the same badge style to keep it consistent! Thanks for that!

image

**Review by Siumauricio — APPROVED** [review]
Thanks you for this!

**Comment by DZsolt01** [comment]
Awesome work! Maybe add a little timeout between searches, if I understood the code correctly it will run at every down so for a "test" word will call 4 times.

**Comment by DJKnaeckebrot** [comment]
> Awesome work! Maybe add a little timeout between searches, if I understood the code correctly it will run at every down so for a "test" word will call 4 times.

Thank you! You are actually right, it will rerender on every key down. I am not too sure if it will retrigger the API call tho, it uses react-query to it should be catched. But I'll add a debounce on the setSearch so it will wait 200ms till it changes the value to cut down on potential rerenders. Thanks for the hint @DZsolt01 !

@Siumauricio I'll implement a patch for this and open a new PR :)

**Pull request merged**